src/core/thread/topology.hpp

```diff
@@ -50,33 +51,11 @@ namespace Thread {
 class Neighbor
 {
 public:
-    Mac::ExtAddress mMacAddr;            ///< The IEEE 802.15.4 Extended Address
-    uint32_t        mLastHeard;          ///< Time when last heard.
-    union
-    {
-        struct
-        {
-            uint32_t mLinkFrameCounter;  ///< The Link Frame Counter
-            uint32_t mMleFrameCounter;   ///< The MLE Frame Counter
-            uint16_t mRloc16;            ///< The RLOC16
-        } mValid;
-        struct
-        {
-            uint8_t mChallenge[Mle::ChallengeTlv::kMaxSize];  ///< The challenge value
-            uint8_t mChallengeLength;    ///< The challenge length
-        } mPending;
-    };
-
-    uint32_t mKeySequence;               ///< Current key sequence
-
     /**
      * Neighbor link states.
      *
      */
     enum State
-#if _WIN32
-        : unsigned int
```

Comment:
Few other related changes:
1. `mState` type was changed to `uint8_t`.
2. `static_cast<>` is used to convert between the `uint8_t` and `State` types.
3. `test_toolchain.cpp` is also updated to validate the `GetState()`/`SetState()` methods.

I would hope these changes avoid the need for the `if _WIN32` pre-processor statement.